src/plasma/WeaklyIonizedGas.cpp

```diff
@@ -0,0 +1,464 @@
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/plasma/WeaklyIonizedGas.h"
+#include "cantera/base/utilities.h"
+#include "cantera/numerics/funcs.h"
+#include <Eigen/SparseLU>
+
+namespace Cantera {
+
+typedef Eigen::Triplet<double> Triplet;
+
+//! Calculate the norm of EEDF
+double norm(const Eigen::VectorXd& f, const vector_fp& grid)
```

Comment:
I think the fact that it's an odd mix of types (`Eigen::vectorXd` and `std::vector`), it's better to just leave it here rather than making it generally available. Possibly worth revisiting later, though.